provider/resource_keycloak_realm_test.go

```diff
@@ -150,6 +150,66 @@ func TestAccKeycloakRealm_themesValidation(t *testing.T) {
 	})
 }
 
+func TestAccKeycloakRealm_InternationalizationValidation(t *testing.T) {
+	realm := "terraform-" + acctest.RandString(10)
+
+	resource.Test(t, resource.TestCase{
+		Providers:    testAccProviders,
+		PreCheck:     func() { testAccPreCheck(t) },
+		CheckDestroy: testAccCheckKeycloakRealmDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config:      testKeycloakRealm_internationalizationValidationWithoutSupportedLocales(realm, "en"),
+				ExpectError: regexp.MustCompile("validation error: SupportLocales should be set if Internationalization is present"),
+			},
+			{
+				Config:      testKeycloakRealm_internationalizationValidation(realm, "en", ""),
+				ExpectError: regexp.MustCompile("validation error: DefaultLocale should be set if Internationalization is present"),
+			},
+			{
+				Config:      testKeycloakRealm_internationalizationValidation(realm, "en", "de"),
+				ExpectError: regexp.MustCompile("validation error: DefaultLocale should be in the SupportLocales"),
+			},
+		},
+	})
+}
+
+func TestAccKeycloakRealm_Internationalization(t *testing.T) {
+	realm := "terraform-" + acctest.RandString(10)
+
+	resource.Test(t, resource.TestCase{
+		Providers:    testAccProviders,
+		PreCheck:     func() { testAccPreCheck(t) },
+		CheckDestroy: testAccCheckKeycloakRealmDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testKeycloakRealm_internationalizationValidation(realm, "en", "en"),
+				Check:  testAccCheckKeycloakRealmInternationalizationIsEnabled("keycloak_realm.realm", "en"),
+			},
```

Comment:
Could you add another test step after this one that updates the internationalization settings to something different?

Reply:
added an extra step that changes the default locale to es